src/test/scala/com/microsoft/hyperspace/index/rules/RuleUtilsTest.scala

```diff
@@ -211,21 +214,21 @@ class RuleUtilsTest extends HyperspaceRuleSuite with SQLHelper {
             optimizedPlan,
             hybridScanEnabled = false,
             hybridScanDeleteEnabled = false,
-            expectCandidateIndex = false,
+            numExpectCandidateIndex = 0,
             expectedHybridScanTag = None,
             expectedCommonSourceBytes = None)
           verify(
             optimizedPlan,
             hybridScanEnabled = true,
             hybridScanDeleteEnabled = false,
-            expectCandidateIndex = false,
+            numExpectCandidateIndex = 0,
             expectedHybridScanTag = None,
             expectedCommonSourceBytes = None)
           verify(
             optimizedPlan,
             hybridScanEnabled = true,
             hybridScanDeleteEnabled = true,
-            expectCandidateIndex = true,
+            numExpectCandidateIndex = 1,
```

Comment:
Question: Is the fix affecting this scenario only or other scenarios as well?

Reply:
Because I always enabled the lineage for all hybrid scan tests, it doesn't affect other test scenarios.
(If there's a such test, it would have detected this bug...)